sstables/mx/reader.cc

```diff
@@ -1369,26 +1401,59 @@ class mx_sstable_mutation_reader : public mp_row_consumer_reader_mx {
         return [this] {
             if (!_index_in_current_partition) {
                 _index_in_current_partition = true;
+                // FIXME reversed multi partition reads
                 return get_index_reader().advance_to(*_current_partition_key);
             }
             return make_ready_future();
-        }().then([this, pos] {
-            return get_index_reader().advance_to(*pos).then([this] {
-                index_reader& idx = *_index_reader;
-                auto index_position = idx.data_file_positions();
-                if (index_position.start <= _context->position()) {
-                    return make_ready_future<>();
-                }
-                return skip_to(idx.element_kind(), index_position.start).then([this, &idx] {
+        }().then([this, pos = *pos] {
+            if (_slice.options.contains(query::partition_slice::option::reversed)) {
+                // the position is a start of a reverse range, so it has its bound_weight reversed
+                // we use the non-reversed schema in the index_reader, so we need to reverse it again
+                bound_weight new_weight = pos.get_bound_weight() == bound_weight::before_all_prefixed ? bound_weight::after_all_prefixed
+                                        : pos.get_bound_weight() == bound_weight::after_all_prefixed ? bound_weight::before_all_prefixed
+                                        : bound_weight::equal;
```

Comment:
Don't we have some `reverse(bound_weight)` now for this?

Reply:
I use it in memtable reader: https://github.com/scylladb/scylla/pull/9174/commits/6eab8baf9acd1885507d487b51d4f457d2fa65ab